src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -558,7 +558,7 @@ private void PreInitialize()
             // taking this snapshot once and reusing at various points during initialization allows us to
             // minimize disk operations
             _directorySnapshot = Directory.EnumerateDirectories(ScriptConfig.RootScriptPath).ToImmutableArray();
-            TraceWriter.Verbose("Created directory snapshot.");
+            TraceWriter?.Verbose("Created directory snapshot.");
```

Comment:
In normal Azure Functions scenarios, the TraceWriter will be defaulted before the host is created, so will be available early in the pipeline. In SelfHost, that is not the case.

Reply:
can you add a test to verify scripthost initialization works when self hosted?